services/src/main/java/org/jboss/windup/web/services/rest/RegisteredApplicationEndpointImpl.java

```diff
@@ -36,23 +47,112 @@
+            throw new NotFoundException();
+        }
+
+        return application;
+    }
+
+    @Override
+    public RegisteredApplication registerApplication(MultipartFormDataInput data, long appGroupId) {
+        Map<String, List<InputPart>> uploadForm = data.getFormDataMap();
+        List<InputPart> inputParts = uploadForm.get("file");
+
+        int uploadedFiles = inputParts.size();
+
+        if (uploadedFiles > 1) {
+            throw new BadRequestException("Use endpoint /multiple to register multiple applications");
+        } else if (uploadedFiles== 0) {
+            throw new BadRequestException("Please provide a file");
+        }
+
+        return this.createApplicationFromInputPart(inputParts.get(0), appGroupId);
+    }
+
+    @Override
+    public Collection<RegisteredApplication> registerMultipleApplications(MultipartFormDataInput data, long appGroupId)
     {
-        String filename = Paths.get(application.getInputPath()).getFileName().toString();
-        Path outputPath = webPathUtil.createWindupReportOutputPath(filename);
-        application.setOutputPath(outputPath.toAbsolutePath().toString());
-        LOG.info("Registering an application at: " + application.getInputPath() + " with output directory: " + outputPath);
+        Map<String, List<InputPart>> uploadForm = data.getFormDataMap();
+        List<InputPart> inputParts = uploadForm.get("file");
 
-        entityManager.persist(application);
+        List<RegisteredApplication> registeredApplications = new ArrayList<>();
 
-        if (application.getApplicationGroup() != null)
-        {
-            if (!application.getApplicationGroup().getApplications().contains(application))
-                application.getApplicationGroup().getApplications().add(application);
+        for (InputPart inputPart : inputParts) {
+            RegisteredApplication application = this.createApplicationFromInputPart(inputPart, appGroupId);
+            registeredApplications.add(application);
+        }
 
-            entityManager.merge(application.getApplicationGroup());
+        return registeredApplications;
+    }
+
+    protected RegisteredApplication createApplicationFromInputPart(InputPart inputPart, long appGroupId) {
+        ApplicationGroup group = this.entityManager.find(ApplicationGroup.class, appGroupId);
+
+        if (group == null) {
+            throw new BadRequestException("Application group not found");
+        }
+
+        try {
+            RegisteredApplication application = new RegisteredApplication();
+
+            MultivaluedMap<String, String> header = inputPart.getHeaders();
+            String fileName = this.getFileName(header);
+
+            //convert the uploaded file to inputstream
+            InputStream inputStream = inputPart.getBody(InputStream.class,null);
+
+            String filePath = Paths.get(this.webPathUtil.getAppPath().toString(), fileName).toString();
+
+            this.saveFileTo(inputStream, filePath);
```

Comment:
What happens if multiple users upload files with the same filename? Or if the same filename gets uploaded to multiple groups? Will it sometimes overwrite the existing one?